src/journal/compress.c

```diff
@@ -307,23 +306,37 @@ int decompress_startswith_lz4(const void *src, uint64_t src_size,
 
         r = LZ4_decompress_safe_partial((char*)src + 8, *buffer, src_size - 8,
                                         prefix_len + 1, *buffer_size);
-        if (r >= 0)
-                size = (unsigned) r;
-        else {
-                /* lz4 always tries to decode full "sequence", so in
-                 * pathological cases might need to decompress the
-                 * full field. */
+        /* One lz4 < 1.8.3, we might get "failure" (r < 0), or "success" where
+         * just a part of the buffer is decompressed. But if we get a smaller
+         * amount of bytes than requested, we don't know whether there isn't enough
+         * data to fill the requested size or whether we just got a partial answer.
+         */
+        if (r < 0 || (unsigned) r < prefix_len + 1) {
```

Comment:
not that it matters, but wouldn't be a cast to `size_t` be nicer than `unsigned` here? I mean, `r` is an `int` and you compare it with a `size_t`, hence why pick `unsigned` which is identical to neither type...

Reply:
Updated with that cast changed.